### PR TITLE
feat(container): add durable startup commands

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?

Adds two new container customization options for registering startup commands that are persisted as script files inside the container's filesystem, so they survive daemon restarts and Docker engine container resurrection. Unlike `WithStartupCommand`, which only fires its in-process PostStarts hook on the `Container` that created the sandbox, the durable variant writes the commands as part of the container's own filesystem state — they are reusable across reconnect paths the SDK does not own.

A single dispatcher script is rendered alongside, at `/etc/durable-startup.d/run.sh`. Invocation is the consumer's responsibility: typically register the dispatcher as a regular `WithStartupCommand` for first-create coverage, and invoke it manually from any reconnect path the consumer owns. The boundary keeps go-sdk out of caller-specific lifecycle semantics — go-sdk renders, the caller invokes.

## Why is it important?

Consumers that compose container behavior from many independent units (plugins, extensions, kits) need a way to ensure their startup commands keep running after a container is resurrected. The existing `WithStartupCommand` only covers the original `Container` instance in process; once the daemon restarts or the container is resurrected, those hooks are gone.

This PR provides the persistence primitive without committing the SDK to any particular invocation policy. Each unit can contribute its own commands under a named namespace; the registration order across the consumer's options is deterministic, so the same composition produces a byte-identical filesystem layout regardless of how independent contributors arrived at the final option list.

## Changes

- New `container/durable_startup.go`. Exports `WithDurableStartupCommand`, `WithDurableStartupCommandsFromDir`, `DurableStartupDir`, `DurableStartupDispatcherPath`, and `ErrDurableStartupReservedNamespace`. Layout is `/etc/durable-startup.d/NNN-<namespace>/MMM-cmd.sh` plus a single `run.sh` dispatcher. Default (dir-less) commands always land at `000-default`.
- The renderer translates `exec.WithWorkingDir`, `exec.WithEnv`, and `exec.WithUser` into POSIX shell directives. `WithUser` wraps the body in ``su -s /bin/sh -c '<body>' '<user>'``; user-missing / `su`-missing failures propagate via `set -e`.
- `WithUser` is validated against a POSIX login-name shape (`^[a-zA-Z_][a-zA-Z0-9_-]*$`) at definition time. Docker's full user spec (`uid:gid`, `user:group`, bare UIDs) is rejected with a clear error pointing the caller at `WithStartupCommand` for full Docker user-spec semantics, since `su` cannot resolve those forms portably across busybox/coreutils.
- The dispatcher is rendered with `LC_ALL=C` so iteration order is byte-deterministic across locales, and walks `*/*-cmd.sh` so unrelated files in a namespace directory are not picked up.
- Namespace allocation is derived from `def.files` alone (no new `Definition` field). Repeated registrations of the same name append to the existing namespace directory; the reserved `default` namespace is always indexed `000`.

## How to test this PR

- `container/durable_startup_test.go` — unit suite covering rendering, namespace resolver, transactional rendering on errors, layout snapshot, determinism, shell-quoting (table-driven), reserved-namespace contract, and the `WithUser` validation surface.
- `container/durable_startup_exec_test.go` — gated with `//go:build !windows`. Executes the rendered scripts under `/bin/sh` and verifies argv byte-exactness, env exporting, working-dir honoring, dispatcher ordering, and `set -e` propagation. Includes a POSIX `sh -n` syntactic check for every rendered file.
- `container/container.run_test.go` — six new `t.Run` subtests of `TestRun` covering the alpine container integration, including four robust `WithUser` cases (single-user switch, env+cwd inside `su`, missing-user failure propagation via dispatcher exit code, per-namespace user isolation) and a layout-on-disk snapshot. Folded into `TestRun` to match the existing `with-startup-command` / `with-after-ready-command` house style.
- `container/container.run_benchmark_test.go` — two new `b.Run` cases under `BenchmarkRun` (`with-durable-startup-command`, `with-durable-startup-command-from-dir`) measuring the option's contribution to `container.Run` end-to-end.
- All new tests pass under `-race`.

## Related issues

- Related to #155 — discovered while writing the missing-user test: `WithStartupCommand` (and `WithAfterReadyCommand`) silently swallow non-zero exit codes from the inner process. Filed separately, not addressed here.

### CODE REVIEW

**Review by kiview — DISMISSED** [review]

File: container/durable_startup.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+	if po.ExecConfig.User != "" {
+		// `su` spawns a fresh shell that does NOT inherit `set -e` from
+		// the outer script, so prepend `set -e` to the inner body. Order
+		// is options-first then USER (POSIX getopt-strict shells stop at
+		// the first non-option arg).
+		inner := "set -e\n" + body.String()
+		fmt.Fprintf(&b, "exec su -s /bin/sh -c %s %s\n",
+			shellSingleQuote(inner),
+			shellSingleQuote(po.ExecConfig.User),
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply container/durable_startup.go:339]: Fixed in aa0ab1e1: WithUser now validates against a POSIX login-name shape (`^[a-zA-Z_][a-zA-Z0-9_-]*$`) at render time and rejects everything else with a clear error pointing at WithStartupCommand for full Docker user-spec semantics.

Accepted the spirit of [the comment](https://github.com/docker/go-sdk/pull/154#discussion_r3208308142) — the silent-runtime-failure footgun is real — but went with fail-loud-at-definition-time rather than bridging `uid:gid` in shell. Bridging properly across busybox/coreutils isn't tractable: `setpriv`/`chroot --userspec=` aren't in alpine, and `su`'s primary-group-only switch loses the gid semantic, so a half-bridge would be more confusing than no bridge.

The error message names the offending value and points to WithStartupCommand explicitly, so callers reusing existing `WithUser("1000:1000")`-style values now see the gap up front instead of debugging a runtime su failure.

Happy to revisit if you see a portable bridging approach I missed.

**Review by robmry — APPROVED** [review]
